### PR TITLE
Add function that returns all paths of an object

### PR DESCRIPTION
**Pull request by sdepold**
Hi Mick,

I hope I'm following the project's style and patterns properly. This new function returns all paths of an object. Not sure if you want something like this in dottie but I needed it and thought it would be a nice addition to the project :)

:wave:

### CODE REVIEW

File: test/paths.test.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+var expect = require("expect.js")
+  , dottie = require("../dottie");
+
+describe("dottie.paths", function() {
+  it("returns an empty array for non-objects", function() {
+    expect(dottie.paths("no object")).to.eql([]);
+  });
```
Comment (mickhansen) [inline test/paths.test.js]: Should perhaps throw instead?

Comment (sdepold) [inline test/paths.test.js]: :+1:

Comment (sdepold) [inline test/paths.test.js]: fixed

**Comment by mickhansen** [comment]
v1.1.0

**Comment by sdepold** [comment]
:bow:

**Pull request merged**